spatialmath/capsule.go

```diff
@@ -112,14 +112,17 @@ func (c *capsule) AlmostEqual(g Geometry) bool {
 // Transform premultiplies the capsule pose with a transform, allowing the capsule to be moved in space.
 func (c *capsule) Transform(toPremultiply Pose) Geometry {
 	newPose := Compose(toPremultiply, c.pose)
+	segB := Compose(toPremultiply, NewPoseFromPoint(c.segB)).Point()
+	center := newPose.Point()
```

Comment:
actually not sure why we are storing this at all given that we have a `Pose`.  The caching speed up has got to be super minor right?

Reply:
Cacheing speedup was minor but definitely noticeable, 4% or so. The collision computations are *the* bottleneck for motion planning, so this is one of the few places it's worth it (others being FK calculations and nlopt)